File: two_pointers/1. Two Sum _ sorted/two_sum.py
def pair_with_targetsum(arr, target_sum):
  arr.sort
  left, right = 0, len(arr) - 1
  while(left < right):
    current_sum = arr[left] + arr[right]
    if current_sum == target_sum:
      return [left, right]

    if target_sum > current_sum:
      left += 1  # we need a pair with a bigger sum
    else:
      right -= 1  # we need a pair with a smaller sum
  return [-1, -1]


def main():
  print(pair_with_targetsum([1, 2, 3, 4, 6], 6))
  print(pair_with_targetsum([2, 5, 9, 11], 11))
  print(pair_with_targetsum([3,2,4], 6))


main()

File: two_pointers/1. Two Sum _ sorted/two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
	left, right = 0, nums.length-1
	while left < right
		current_sum = nums[left] + nums[right]
		if current_sum == target
			return [left, right]
		end
		puts "current_sum : #{current_sum}"

		if target > current_sum
			left += 1
		else
			right -= 1
		end
	end

	return [-1,-1]

end



nums = [2,7,11,15]
target = 9

nums = [3,2,4]
target = 6

res = two_sum(nums, target)
puts "res : #{res}"